src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs

```diff
@@ -86,41 +87,83 @@ public HttpConnectionSettings CloneAndNormalize()
                 _cookieContainer = new CookieContainer();
             }
 
-            return new HttpConnectionSettings()
+            // TODO: Replace with Platform-Guard Assertion Annotations once https://github.com/dotnet/runtime/issues/44922 is finished
+            // TODO: Remove if/when QuicImplementationProvider is removed from System.Net.Quic.
+            if ((OperatingSystem.IsLinux() && !OperatingSystem.IsAndroid()) || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS())
```

Comment:
You don't need to duplicate the code if you do something like

```c#
var hts = HttpConnectionSettings() {
// common init
};

if ((OperatingSystem.IsLinux() && !OperatingSystem.IsAndroid()) || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS())
	hts. _quicImplementationProvider = _quicImplementationProvider;

return hts;
```

Reply:
Nice, will do.